src/driver/odoc_unit.ml

```diff
@@ -34,109 +51,186 @@ type mld = [ `Mld ]
-          | None -> None
-          | Some (pkg, lib) -> Some (dash_l pkg lib))
-        libraries
-    in
-    { pages; libs }
+    let libs_rel = List.concat_map dash_l libraries in
+    let map_rel dir = List.map (fun (a, b) -> (a, Fpath.(dir // b))) in
+    let pages = map_rel output_dir pages_rel in
+    let libs = map_rel output_dir libs_rel in
+    let pages_linked = map_rel linked_dir pages_rel in
+    let libs_linked = map_rel linked_dir libs_rel in
+    { pages; libs; pages_linked; libs_linked }
   in
-  let pkg_args =
+  let args_of =
     let cache = Hashtbl.create 10 in
-    fun pkg : pkg_args ->
-      match Hashtbl.find_opt cache pkg with
+    fun pkg lib_deps : pkg_args ->
+      match Hashtbl.find_opt cache (pkg, lib_deps) with
       | Some res -> res
       | None ->
-          let { pages = own_page; libs = own_libs } = pkg_args_of_pkg pkg in
-          let { pages = config_pages; libs = config_libs } =
-            pkg_args_of_config pkg.Packages.config
+          let {
+            pages = own_page;
+            libs = own_libs;
+            pages_linked = own_p_l;
+            libs_linked = own_l_l;
+          } =
+            base_args pkg lib_deps
+          in
+          let {
+            pages = config_pages;
+            libs = config_libs;
+            pages_linked = cfg_p_l;
+            libs_linked = cfg_l_l;
+          } =
+            args_of_config pkg.Packages.config
           in
-          { pages = own_page @ config_pages; libs = own_libs @ config_libs }
+          let result =
+            {
+              pages = own_page @ config_pages;
+              libs = own_libs @ config_libs;
+              pages_linked = own_p_l @ cfg_p_l;
+              libs_linked = own_l_l @ cfg_l_l;
+            }
+          in
+          Hashtbl.add cache (pkg, lib_deps) result;
+          result
   in
 
   let index_of pkg =
-    let pkg_args = pkg_args_of_pkg pkg in
+    let pkg_libs =
+      List.map (fun l -> l.Packages.lib_name) pkg.Packages.libraries
+      |> Util.StringSet.of_list
+    in
+    let pkg_args = base_args pkg pkg_libs in
     let output_file = Fpath.(index_dir / pkg.name / Odoc.index_filename) in
     { pkg_args; output_file; json = false; search_dir = pkg.pkg_dir }
   in
 
-  let make_unit ~name ~kind ~rel_dir ~input_file ~pkg ~include_dirs : _ unit =
+  let make_unit ~name ~kind ~rel_dir ~input_file ~pkg ~include_dirs ~lib_deps :
+      _ unit =
     let ( // ) = Fpath.( // ) in
     let ( / ) = Fpath.( / ) in
+    let pkg_args = args_of pkg lib_deps in
     let odoc_dir = output_dir // rel_dir in
     let parent_id = rel_dir |> Odoc.Id.of_fpath in
-    let odoc_file = odoc_dir / (name ^ ".odoc") in
-    let odocl_file = linked_dir // rel_dir / (name ^ ".odocl") in
+    let odoc_file = odoc_dir / (String.uncapitalize_ascii name ^ ".odoc") in
+    (* odoc will uncapitalise the output filename *)
+    let odocl_file =
+      linked_dir // rel_dir / (String.uncapitalize_ascii name ^ ".odocl")
```

Comment:
I'm not sure this is a reasonable things for Odoc to do, as capitalization rules became more complex in https://github.com/ocaml/ocaml/pull/12664.
Though, it's reasonable to do this in the driver as long as Odoc does this.

Reply:
Mmm, odoc _is_ currently doing it, which is why the driver needed fixing here. I'm not 100% convinced it's the best way, but I'd rather fix that in a separate PR, if necessary.